geoapps/inversion/base_inversion_application.py

```diff
@@ -1472,6 +1460,52 @@ def get_inversion_params_dict(
         param_dict["window_azimuth"] = 0.0
         return param_dict
 
+    def detrend_data(
+        self,
+        param_dict: dict,
+        workspace: Workspace,
+        detrend_order: int,
+        detrend_type: str,
+    ) -> dict:
+        """
+        Detrend data and update data values in param_dict.
+
+        :param param_dict: Dictionary of params to create self._run_params.
+        :param workspace: Output workspace.
+        :param detrend_order: Order of the polynomial to be used.
+        :param detrend_type: Method to be used for the detrending.
+            "all": Use all points.
+            "perimeter": Only use points on the convex hull .
+
+        :return: Updated param_dict with updated data.
+        """
+        if detrend_type != "none" and detrend_order is not None:
```

Comment:
Do reverse logic and return fast

```suggestion
        if not (detrend_type != "none" and detrend_order is not None):
            return param_dict
```

Reply:
Remove indentation below